### PR TITLE
Tutorial for adding new sensors

### PR DESCRIPTION
**Pull request by nsubiron**
Adding a tutorial to the docs explaining how the sensor system works and how to implement a new sensor.

---
This change is [Reviewable](https://reviewable.io/reviews/carla-simulator/carla/1884)

### CODE REVIEW

File: Docs/dev/how_to_add_a_new_sensor.md [hunk 1]
```diff
@@ -0,0 +1,569 @@
...
+Thus we'll need the following classes covering the different steps of the
+pipeline
+
+  * **Sensor actor**<br>
+    Actor in charge of measuring and/or simulating data. Running in Carla plugin
+    using UE4 framework. Accessible by the user as Sensor actor.
+
+  * **Serializer**<br>
+    Object containing methods for serializing and deserializing the data
+    generated by the sensor. Running in LibCarla, both server and client.
```
Comment (ThrashAbaddon) [inline Docs/dev/how_to_add_a_new_sensor.md]: In which format is data serialized?

File: Docs/dev/how_to_add_a_new_sensor.md [hunk 2]
```diff
@@ -0,0 +1,569 @@
...
+    that will be passed to the final user, both in C++ and Python APIs.
+
+!!! note
+    To ensure best performance, sensors are registered and dispatched using a
+    sort of "compile-time plugin system" based on template meta-programming.
+    Most likely, the code won't compile until all the pieces are present.
+
+## Creating a new sensor
+
+[**Full source code here.**](https://gist.github.com/nsubiron/011fd1b9767cd441b1d8467dc11e00f9)
```
Comment (ThrashAbaddon) [inline Docs/dev/how_to_add_a_new_sensor.md]: Maybe source code for example could be in this repository under clearly named folder structure?

File: Docs/dev/how_to_add_a_new_sensor.md [hunk 3]
```diff
@@ -0,0 +1,569 @@
...
+## Appendix: Client-side sensors
+
+Some sensors do not require the simulator to do their measurements, those
+sensors may run completely in the client-side freeing the simulator from extra
+computations. Examples of such sensors are the _GNSS_ and the _LaneInvasion_
+sensors.
+
+The usual approach is to create a "dummy" sensor in the server-side, just so the
+simulator is aware that such actor exists. However, this dummy sensor does tick
+nor sends any sort of data. Its counterpart on the client-side however,
```
Comment (ThrashAbaddon) [inline Docs/dev/how_to_add_a_new_sensor.md]: Was the intention here maybe to write
>However, this dummy sensor `doesn't` tick nor sends any sort of data.

?

File: Docs/dev/how_to_add_a_new_sensor.md (hunk 1 above)
Comment (nsubiron) [reply Docs/dev/how_to_add_a_new_sensor.md]: That's up to the sensor, that's why you need to provide serialize/desarialize methods, they should convert the data to a buffer object, the binary data in the buffer is sent as it is.

File: Docs/dev/how_to_add_a_new_sensor.md (hunk 3 above)
Comment (nsubiron) [reply Docs/dev/how_to_add_a_new_sensor.md]: Yes! thanks I'll fix that

**Pull request merged**